guides/intro.md

```diff
@@ -0,0 +1,615 @@
+in the previous section), the server, or both depending on when the peer joined. Also, don't forget that after you add or remove tracks from a PeerConnection,
+new negotiation has to take place and is signaled by a `{:ex_webrtc, _from, :negotiation_needed}` message from the PeerConnection process!
+
+> #### The caveats of negotiation {: .tip}
+> But wait, the peer who added new tracks doesn't have to start the negotiation?
+>
+> Certainly, that's the simplest way, but as long as the *number of transceivers* of the offerer (or the number of mlines in the offer SDP with appropariate `direction`
+> attribute set to be specific) is greater or
+> equal to the number of all tracks added by the answerer, the tracks will be considered in the negotiation.
+>
+> But what does that even mean?
+> Each transceiver is responsible for sending and/or receiving a single track. When you call `PeerConnection.add_track`, we actually look for a free transceiver
+> (that is, one that is not sending a track already) and use it, or create a new transceiver in case we do not find anything suitable. If you are very sure
+> that the remote peer added _N_ new video tracks, you can add _N_ video transceivers (using `PeerConnection.add_transceiver`) and begin the negotiation as
+> the offerer. If you didn't add the transceivers, the tracks added by the remote peer (the answerer) would be ignored.
+
+Let's look at an example:
+1. The first peer (Peer 1) joins - here it probably makes more sense for the client (so the Web Browser) to start the negotiation, as the server (Elixir App/
+`Forwarder` in the diagram) does not know how many tracks the client wants to add (the `2. offer/answer` message indicates exchange of offer where the direction of
+the arrow means the direction of the offer message).
+
+```mermaid
+flowchart LR
+  subgraph P1["Peer 1 (Web Browser)"]
+    User-- "1. addTrack(track11)" -->PCW1[PeerConnection]
+  end
+
+  subgraph elixir [Elixir App]
+    PCE1[PeerConnection 1]-- "3. {:track, track11}" -->Forwarder
+  end
+
+  PCW1-. "2. offer/answer" .->PCE1
+```
+
+2. The second peer (Peer 2) joins - now we need to make a decision: we want Peer 2 to receive track from Peer 1, but Peer 2 also wants to send some tracks.
+We can either:
+    - perform two negotiations: first, where Peer 2 is the offerer and adds their tracks, second where the server is the offerer and adds Peer 1's tracks to Peer 2's PeerConnection.
+
+    ```mermaid
+    flowchart LR
+      subgraph elixir [Elixir App]
+        PCE1[PeerConnection 1]
+        Forwarder-- "4. add_track(track12)" -->PCE2[PeerConnection 2]
+        PCE2-- "3. {:track, track22}" -->Forwarder
+      end
+
+      subgraph P2["Peer 2 (Web Browser)"]
+        U2[User]-- "1. addTrack(track22)" -->PCW2[PeerConnection]
+        PCW2-- "6. ontrack(track12)" --> U2
+      end
+
+      PCW2-. "2. offer/answer" .->PCE2
+      PCE2-. "5. offer/answer" .->PCW2
+    ```
```

Comment:
I think hand-sculpting the diagrams in Excalidraw is a must. In mermaid everything is floating around and makes the steps harder to comprehend, as you first need to adapt to the different positions of the same things from above